### ISSUE TITLE
Download a film or series

### ISSUE THREAD

**Issue by a1986u** [recloudstream/cloudstream#914]
### Describe your suggested feature

I can watch TV series in Turkish. I say change the language from the source and select Turkish and I can watch it this way. But when I want to download TV series or movies, this option does not appear and it downloads in English every time. I think it's a problem that needs to be fixed. And I want to select download quality if exist. 

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by a1986u** [comment]
Yes that's right. It would be great if we could choose the resolution and audio language before downloading.

**Comment by asl97** [comment]
> Yes that's right. It would be great if we could choose the resolution and audio language before downloading.

You can already choose the resolution using `long press > download mirror` if the provider support it.

or are you requesting for reencoding after downloading videos?

I honestly wouldn't expect multitrack support any time soon as other provides uses separate sources for audio, for example, one for raw and another for dub, it just is not common for multitrack.

A workaround is using the `long press > copy link` and using your choice of video downloader, like youtube-dl or one of its forks.

```bash
yt-dlp -F 'link here'
[generic] Extracting URL:  ...
[generic] master.m3u8...
Downloading webpage
[generic] master.m3u8...
Downloading m3u8 information
[generic] master.m3u8...
Checking m3u8 live status
[info] Available formats for master.m3u8...
ID             EXT RESOLUTION FPS │   FILESIZE   TBR PROTO │ VCODEC        VBR ACODEC     MORE INFO
──────────────────────────────────────────────────────────────────────────────────────────────────────
audio0-English mp4 audio only     │                  m3u8  │ audio only        unknown    [en] English
audio0-Türkçe  mp4 audio only     │                  m3u8  │ audio only        unknown    [tr] Türkçe
1444           mp4 1280x720    24 │ ~438.18MiB 1444k m3u8  │ avc1.64001f 1444k video only
```

**Comment by a1986u** [comment]
Thank you very much. I can download with 1DM downloader. Long press > copy link and download with 1DM downloader.
